### DISCUSSION TITLE
Raw text styling?

### DISCUSSION THREAD

**Discussion by alexpdp7** [typst/typst#1228] [Ideas]
Hi, I saw [you asking for feedback](https://twitter.com/typstapp/status/1658527682353561619?s=20), so I thought I'd ask...

Secondary question: my team is "abandoning" print; is there any plan to produce epub, HTML, etc.?

Main question: we write courses about systems administration. We provide lots of code blocks of stuff that the student must type. We currently use AsciiDoc, where we can enable markup inside code blocks. We use this to highlight "user input" (what the student must type), "replaceable" (stuff where the student must fill a custom value, e.g. depending on the output of a previous command), and highlighting stuff.

Is there a plan to add some similar functionality? Or is there any comfy way to do this already? This would have all the "requirements" of a raw block; preserving whitespace and linebreaks, etc.

(This also introduces challenges; underscores, asterisks, etc. may appear in the code block.)

Cheers,

Álex

**Comment by maxcrees** [comment]
I think this is certainly achievable right now. You can arbitrarily style content within a raw block. For example, you might write a show rule that uses regex to look for and style certain parts.

I can whip up a demo tomorrow if you like.

**Reply by maxcrees** [reply]
As for HTML - you might search other discussions / issues for that; it's been discussed before. I'm not sure about EPUB however. Someone else might speak to the current situation of those things.

**Reply by alexpdp7** [reply]
Oh, epub "is" HTML, so I'll dig for that. My primary question was about styling on raw blocks. If this is solvable by looking at the docs you may resolve this.

Although we normally do manual styling, stuff like this on AsciiDoc:

```
[subs=+quotes]
[user@host] *cat /etc/file_number_*
_... output omitted..._
bit
bit
this is the *interesting* part
_... output omitted..._
```

Asterisks to differentiate what you must type. The "number" in underscores you need to replace with the result of a previous command. We emphasize output omitted to make it clear this is a "comment". Plus, the highlighting.

It's ad-hoc/explicit, we really can't regex that.

**Reply by laurmaedje** [reply]
Within the raw block, you can't use Typst syntax, but you could parse your own ad-hoc format if it's only light stuff.
